src/V3ReuseWideTemps.cpp

```diff
@@ -0,0 +1,246 @@
+// -*- mode: C++; c-file-style: "cc-mode" -*-
+//*************************************************************************
+// DESCRIPTION: Verilator: Reuse wide temporaries created by V3Premit
+//
+// Code available from: https://verilator.org
+//
+//*************************************************************************
+//
+// Copyright 2003-2024 by Wilson Snyder. This program is free software; you
+// can redistribute it and/or modify it under the terms of either the GNU
+// Lesser General Public License Version 3 or the Perl Artistic License
+// Version 2.0.
+// SPDX-License-Identifier: LGPL-3.0-only OR Artistic-2.0
+//
+//*************************************************************************
+// Optimize wide temporaries in cases when V3Expand cannot handle them,
+// ie. for wide temporaries are not expanded due to their width exceeding
+// expandLimit.
+//
+// Note: this phase requires temporaries in a single static assignment form (SSA).
```

Comment:
```suggestion
// This phase requires temporaries in a single static assignment form (SSA).
```

I'll review rest of this file after it renames, so my comments aren't lost.